src/Pact/ApiReq.hs

```diff
@@ -296,7 +296,7 @@ loadSigData fp = do
     Left e -> Left $ "Error loading SigData file " <> fp <> ": " <> show e
     Right sd -> Right sd
 
-addSigToSigData :: SomeKeyPair -> SigData a -> IO (SigData a)
+addSigToSigData :: Ed25519KeyPair -> SigData a -> IO (SigData a)
```

Comment:
If we start integrating more with other changes, would we still want to monomorphize the signature scheme?

Reply:
There are some integrations we could make that would introduce new PPK schemes. Depending on the details of how those integrations work, they might not see the changes here in `Pact.ApiReq`, since `Pact.ApiReq` is used client-side, e.g. by users signing requests and making API calls to a pact server.

A new integration that merely needs signature verification in pact will not be bothered by the monomorphization happening in client code here. But an integration that also wanted client-side support would indeed be less convenient to write, after we remove `SomeKeyPair`.